build.sbt

```diff
@@ -1,84 +1,138 @@
-      }
-      classLoader.loadClass("me.jeffmay.neo4j.client.SetupBeforeTests").newInstance()
-  },
-  Tests.Cleanup {
-    classLoader: ClassLoader =>
-      classLoader.loadClass("me.jeffmay.neo4j.client.CleanupAfterTests").newInstance()
-  }
-)
+  // Avoid sbt warnings for these libraries
+  dependencyOverrides := Set(
+    "com.typesafe.play" %% "play-json" % playVersion,
+    "com.google.guava" % "guava" % "18.0",
+    "org.scalactic" %% "scalactic" % scalatestVersion
+  ),
+
+  // Don't publish tests.jar
+  publishArtifact in Test := false,
+  // Don't compile or publish ScalaDoc
+  // (ScalaDoc compiler breaks on links and the source code comments are more valuable)
+  sources in(Compile, doc) := Seq.empty,
+  publishArtifact in (Compile, packageDoc) := false
+
+) ++
+  // Publish to https://bintray.com/jeffmay/maven
+  bintraySettings ++ bintrayPublishSettings
+
+lazy val core = (project in file("core")).settings(commonSettings ++ Seq(
+  name := "neo4j-scala-models"
+))
+
+lazy val testUtil = Project("test-util", file("test-util")).settings(commonSettings ++ Seq(
+  name := "neo4j-scala-test-util",
+
+  libraryDependencies ++= Seq(
+    "org.scalatest" %% "scalatest" % scalatestVersion
+  )
+)).dependsOn(core)
+
+lazy val ws = (project in file("ws")).settings(commonSettings ++ Seq(
+
+  name := "neo4j-scala-client",
+
+  libraryDependencies ++= Seq(
+    "com.typesafe.play" %% "play-ws" % playVersion
+  ),
+
+  // Cleanup after tests by default
+  cleanup := true,
+
+  // Call setup and teardown hooks before and after any tests run
+  testOptions ++= Seq(
+    Tests.Setup {
+      classLoader: ClassLoader =>
+        val cleanDBAfterTests = cleanup.value
+        sys.props += "NEO4J_TEST_CLEANUP_AFTER" -> cleanDBAfterTests.toString
+        if (cleanDBAfterTests) {
+          println(AnsiColor.GREEN +
+            """All tests will cleanup their changes to their namespace in the database after they finish.
+              |Note: The tests will cleanup their own namespace from previous test runs before they run.""".stripMargin
+            + AnsiColor.RESET)
+        }
+        else {
+          println(AnsiColor.BLUE +
+            """Tests will NOT cleanup their changes to their namespace in the database after they finish.
+              |Note: The tests will cleanup their own namespace from previous test runs before they run.""".stripMargin
+            + AnsiColor.RESET)
+        }
+        classLoader.loadClass("me.jeffmay.neo4j.client.SetupBeforeTests").newInstance()
+    },
+    Tests.Cleanup {
+      classLoader: ClassLoader =>
+        classLoader.loadClass("me.jeffmay.neo4j.client.CleanupAfterTests").newInstance()
+    }
+  )
+)).dependsOn(core, testUtil % "test", wsTestUtil % "test", wsUtil)
+
+lazy val wsUtil = Project("ws-util", file("ws-util")).settings(commonSettings ++ Seq(
+  name := "neo4j-scala-ws-util",
+
+  libraryDependencies ++= Seq(
+    "com.typesafe.play" %% "play-ws" % playVersion
+  )
+))
```

Comment:
WS code that is used by `ws` source code implementation and `wsTestUtil` test utilities.